hazelcast-sql/src/test/java/com/hazelcast/sql_slow/MapIndexScanPMigrationStressTest.java

```diff
@@ -99,62 +99,74 @@ public void stressTest_sorted() throws InterruptedException {
         IndexConfig indexConfig = new IndexConfig(IndexType.SORTED, "this").setName(randomName());
         map.addIndex(indexConfig);
 
-        MutatorThread mutator = new MutatorThread(1000L);
-        mutator.start();
-
-        assertRowsOrdered("SELECT * FROM " + MAP_NAME + " ORDER BY this DESC", expected);
+        MutatorThread mutator = new MutatorThread(2000L);
+        assertRowsOrdered("SELECT * FROM " + MAP_NAME + " ORDER BY this DESC", expected, mutator);
 
         mutator.terminate();
         mutator.join();
+        assertThat(mutatorException.get()).isNull();
     }
 
     private class MutatorThread extends Thread {
-
-        private volatile boolean active = true;
+        private boolean firstLaunch = true;
+        private boolean active = true;
         private final long delay;
 
         private MutatorThread(long delay) {
             this.delay = delay;
         }
 
-        private void terminate() {
+        private synchronized void terminate() {
```

Comment:
I assume this one protects the `active` field. 
In run method, we read `active` field without locking. Is it going to cause problem?